### PR TITLE
Configure communication between frontend and server in development and when deployed.

### LINKED ISSUE #29: The frontend and server do not communicate with each other.
Currently, we only run one dev server at a time. We need some way to work with both.

### PR DESCRIPTION
**Pull request by JosephBushagour**
Configured the services of our app, and the URL routing between them when deployed. Also configured the frontend and backend development servers to communicate when both running (the run-dev.sh script will run both servers, and halt both servers on Ctrl+C). Small changes to the frontend to enable the ability to refresh the page without getting a 404 (in production and development). Finally, updated the necessary README files with the changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,9 +2,45 @@
...
 ## Code Locations
-* **Server Code**, built in Java Servelets, lives in /server
+* **Server Code**, built in Java Servlets, lives in /server
 * **Web Client Code**, built in React, lives in /frontend
 
+
+## Running Locally
+Both the frontend and backend development servers can be run together
+through the command `bash run-dev.sh` (assuming that one has installed
+all frontend packages with `npm install` and has set their gcloud project
```
Comment (KeatonTech) [inline README.md]: If you want an extension project, you could add a shell script that does this part as well. That'll speed up the process of switching branches.

File: run-dev.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Runs both dev servers at the same time, can be ended at the same time with ctrl+C
```
Comment (KeatonTech) [inline run-dev.sh:1]: Awesome, good idea!

File: run-dev.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+echo "Preview on http://localhost:9000"
+
+trap "kill %1" SIGINT
+(cd frontend/; npm run start-dev) | tee frontend.log | sed -e "s/^/\x1b[32m[FRONTEND]\x1b[0m /" \
```
Comment (KeatonTech) [inline run-dev.sh]: If they're already going to different logfiles, why do the lines need to be prefixed?

**Review by KeatonTech — APPROVED** [review]

File: run-dev.sh (hunk 3 above)
Comment (JosephBushagour) [reply run-dev.sh]: The colored prefix helps for distinguishing the different sources in the terminal when they're running.  Especially when things are scrolling by fast, it's a nice way to visually distinguish what's happening (hopefully this image shows what I mean).
image

Comment (KeatonTech) [reply run-dev.sh]: Ohh, for the STDOUT log. Gotcha yeah that makes sense

**Review by jimmycolema — APPROVED** [review]
lgtm!

File: README.md (hunk 1 above)
Comment (JosephBushagour) [reply README.md]: That's a great idea! I've modified run-dev.sh to take in an optional `-setup` (optionally, just `-s`) flag which runs these one-time commands.

File: frontend/README.md [hunk 4]
```diff
@@ -1,53 +1,45 @@
...
 
-## Usage
-**Development**
 
-`yarn run start-dev`
+## Running Locally
+Run the dev server with the command `npm run start-dev`
 
-* Build app continuously (HMR enabled)
-* App served @ `http://localhost:8080`
```
Comment (SarahGB) [inline frontend/README.md:26]: Formatting it like this might be a bit easier: To use, visit http://localhost:9000

**Review by SarahGB — APPROVED** [review]

File: frontend/README.md (hunk 4 above)
Comment (JosephBushagour) [reply frontend/README.md:26]: That's a good point! I've changed all references to to ports to follow that style.

**Pull request merged**